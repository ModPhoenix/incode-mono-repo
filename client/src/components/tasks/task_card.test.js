import { MemoryRouter } from 'react-router';
import TaskCard from './task_card';

const mockTask = {
  status: 'To do',
  _id: '5b39e7e11deb0934993043cc',
  title: 'Mock Title',
  description: 'Mock Description',
};

it('render correctly', () => {
  const wrapper = shallow(
    <TaskCard
      data={mockTask}
      editTask={() => {}}
      deleteTask={() => {}}
      isSuperuser
    />,
  );

  expect(wrapper).toMatchSnapshot();
});

it('displaying data correctly', () => {
  const wrapper = mount(
    <MemoryRouter>
      <TaskCard
        data={mockTask}
        editTask={() => {}}
        deleteTask={() => {}}
        isSuperuser
      />
    </MemoryRouter>,
  );

  const id = wrapper.find('.task-id').first().text();
  const title = wrapper.find('.task-title').first().text();
  const description = wrapper.find('.task-description').first().text();


  expect(id).toEqual('ID: 5b39e7e11deb0934993043cc');
  expect(title).toEqual('Mock Title');
  expect(description).toEqual('Mock Description');
});
