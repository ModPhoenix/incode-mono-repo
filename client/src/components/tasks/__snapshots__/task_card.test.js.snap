// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<WithStyles(Card)
  className="task-card"
>
  <div
    className="task-card-view"
  >
    <WithStyles(Typography)
      className="task-id"
      gutterBottom={true}
      variant="caption"
    >
      ID:
       
      5b39e7e11deb0934993043cc
    </WithStyles(Typography)>
    <WithStyles(CardContent)
      className="task-content"
    >
      <Link
        className="task-link"
        replace={false}
        to="/task/5b39e7e11deb0934993043cc"
      >
        <WithStyles(Typography)
          className="task-title"
          component="h2"
          gutterBottom={true}
          variant="headline"
        >
          Mock Title
        </WithStyles(Typography)>
      </Link>
      <WithStyles(Typography)
        className="task-description"
        component="p"
      >
        Mock Description
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
    <WithStyles(CardActions)>
      <WithStyles(FormControl)
        className="task-status"
      >
        <WithStyles(Select)
          className="task-status-select"
          displayEmpty={true}
          name="task_status"
          onChange={[Function]}
          value="To do"
        >
          <WithStyles(MenuItem)
            key="To do"
            value="To do"
          >
            To do
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
            key="In Progress"
            value="In Progress"
          >
            In Progress
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
            key="Peer Review"
            value="Peer Review"
          >
            Peer Review
          </WithStyles(MenuItem)>
          <WithStyles(MenuItem)
            key="Completed"
            value="Completed"
          >
            Completed
          </WithStyles(MenuItem)>
        </WithStyles(Select)>
      </WithStyles(FormControl)>
    </WithStyles(CardActions)>
  </div>
  <div
    className="task-container-btn"
  >
    <WithStyles(Button)
      color="primary"
      onClick={[Function]}
      size="small"
      variant="contained"
    >
      Edit
    </WithStyles(Button)>
    <WithStyles(Button)
      color="secondary"
      onClick={[Function]}
      size="small"
      variant="contained"
    >
      Delete
    </WithStyles(Button)>
  </div>
</WithStyles(Card)>
`;
